app/models/user.rb

```diff
@@ -1273,22 +1273,36 @@ def user_fields(field_ids = nil)
     end
   end
 
-  def public_user_field_values
-    public_user_fields.values.join(" ")
+  def validatable_user_fields_values
+    validatable_user_fields.values.join(" ")
   end
 
   def set_user_field(field_id, value)
     custom_fields["#{USER_FIELD_PREFIX}#{field_id}"] = value
   end
 
+  def clean_user_value(value)
+    PrettyText.cook(value).gsub(/^<p>(.*)<\/p>$/, "\\1")
+  end
+
   def apply_watched_words
-    public_user_fields.each do |id, value|
-      set_user_field(id, PrettyText.cook(value).gsub(/^<p>(.*)<\/p>$/, "\\1"))
+    validatable_user_fields.each do |id, value|
+      cooked_value =
+        if value.class == Array
```

Comment:
Since `validatable_user_fields` no longer contains multiselect fields, do we still need to process array values here? There shouldn't be any... right?

Reply:
No longer a need right now, but we can deal with if/when that changes since we now have tests that will catch the missing handling it. Rolling back to the simpler existing version.